modules/swagger-core/src/main/java/io/swagger/core/filter/SpecFilter.java

```diff
@@ -4,314 +4,120 @@
-            if (model != null) {
-                locateReferencedDefinitions(model.getProperties(), nestedReferencedDefinitions, swagger);
+    private OpenAPI filterOpenAPI(OpenAPISpecFilter filter, OpenAPI openAPI, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
+        if (openAPI != null) {
+            Optional<OpenAPI> filteredOpenAPI = filter.filterOpenAPI(openAPI, null, params, cookies, headers);
+            if (filteredOpenAPI.isPresent()) {
+                return filteredOpenAPI.get();
             }
         }
+        return null;
     }
 
-    public Map<String, Schema> filterDefinitions(SwaggerSpecFilter filter, Map<String, Schema> definitions, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
-        if (definitions == null) {
-            return null;
-        }
-        Map<String, Schema> clonedDefinitions = new LinkedHashMap<String, Schema>();
-
-        for (String key : definitions.keySet()) {
-            Schema definition = definitions.get(key);
-            Map<String, Schema> clonedProperties = new LinkedHashMap<String, Schema>();
-            if (definition.getProperties() != null) {
-                for (String propName : (Set<String>)definition.getProperties().keySet()) {
-                    Schema property = (Schema)definition.getProperties().get(propName);
-                    if (property != null) {
-                        boolean shouldInclude = filter.isPropertyAllowed(definition, property, propName, params, cookies, headers);
-                        if (shouldInclude) {
-                            clonedProperties.put(propName, property);
+    private Operation filterOperation(OpenAPISpecFilter filter, Operation operation, String resourcePath, String key, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
+        if (operation != null) {
+            ApiDescription description = new ApiDescription(resourcePath, key);
+            Optional<Operation> filteredOperation = filter.filterOperation(operation, description, params, cookies, headers);
+            if (filteredOperation.isPresent()) {
+                List<Parameter> filteredParameters = new ArrayList<>();
+                Operation filteredOperationGet = filteredOperation.get();
+                List<Parameter> parameters = filteredOperationGet.getParameters();
+                if (parameters != null) {
+                    for (Parameter parameter : parameters) {
+                        Parameter filteredParameter = filterParameter(filter, operation, parameter, resourcePath, key, params, cookies, headers);
+                        if (filteredParameter != null) {
+                            filteredParameter.setSchema(filterProperty(filter, parameter.getSchema(), resourcePath, key, params, cookies, headers));
+                            filteredParameters.add(filteredParameter);
```

Comment:
see also general review comment, filterProperty should be used for schemas inside components also